homeassistant/components/airzone_cloud/climate.py

```diff
@@ -290,9 +314,58 @@ def __init__(
         ]
         if HVACMode.OFF not in self._attr_hvac_modes:
             self._attr_hvac_modes += [HVACMode.OFF]
+        if (
+            self.get_airzone_value(AZD_SPEED) is not None
+            and self.get_airzone_value(AZD_SPEEDS) is not None
+        ):
+            self._initialize_fan_speeds()
 
         self._async_update_attrs()
 
+    def _initialize_fan_speeds(self) -> None:
+        """Initialize Aidoo fan speeds."""
+        azd_speeds: dict[int, int] = self.get_airzone_value(AZD_SPEEDS)
+        max_speed = max(azd_speeds)
+
+        fan_speeds: dict[int, str]
+        if speeds_map := FAN_SPEED_MAPS.get(max_speed):
+            fan_speeds = speeds_map
+        else:
+            fan_speeds = {}
+
+            for speed in azd_speeds:
+                if speed == 0:
+                    fan_speeds[speed] = FAN_AUTO
+                else:
+                    fan_speeds[speed] = f"{int(round((speed * 100) / max_speed, 0))}%"
+
+            fan_speeds[1] = FAN_LOW
+            fan_speeds[int(round((max_speed + 1) / 2, 0))] = FAN_MEDIUM
+            fan_speeds[max_speed] = FAN_HIGH
```

Comment:
Why do we want the list to be:
```
[
        FAN_AUTO,
        FAN_LOW,
        "40%",
        FAN_MEDIUM,
        "80%",
        FAN_HIGH,
    ]
```

Instead of:
```
[
        FAN_AUTO,
        "20 %",
        "40 %",
        "60 %",
        "80 %",
        "100 %",
    ]
```

If there's a good reason, it wouldn't hurt to explain it in a comment.

Reply:
Because only the default speeds are exposed via Homekit/Alexa, and this is the only way of having N speeds and also getting the default ones exposed.

Reply:
Done in 0f4566fd7b73df8542737d0ae8a6c9adc66bc820

Reply:
> only the default speeds are exposed via Homekit/Alexa

Integrations should not have to solve limitations in voice assistant integrations.
Moreover, the Alexa integration does not support fan speeds for climate entities.

For homekit there is such a limitation though
https://github.com/home-assistant/core/blob/9b1180734a7cfddfe0264591fb00b94a33d557f7/homeassistant/components/homekit/type_thermostats.py#L313-L317

Reply:
> I'd like a second opinion on this design, integrations should not have to solve limitations in voice assistant integrations.

@emontnemery this was already proposed by @bdraco during the same review for the local Airzone integration, see https://github.com/home-assistant/core/pull/92840?w=1#discussion_r1191986188
The Airzone local integration code: https://github.com/home-assistant/core/blob/7c4747bb02e07a126728dbb8fbf5005257a50c9c/homeassistant/components/airzone/climate.py#L158-L177

Reply:
> The Alexa integration does not support fan speeds for climate entities.

You're right, comment fixed in 681967f516b31703c9114f6a36548a20a2215530

Reply:
> this was already proposed by @bdraco during the same review for the local Airzone integration

That's not a good proposal though, other integrations should not have to implement workarounds because of limitations in `homekit`, please change to a consistent naming both in this PR and in the `airzone` integration.

Reply:
> That's not a good proposal though, other integrations should not have to implement workarounds because of limitations in `homekit`, please change to a consistent naming both in this PR and in the `airzone` integration.

Done in 438c5b6e32483f276b47cb0d5e4747ee5b031d78.
I don't think that doing it for Airzone integration is a good idea since it will break the compatibility for existing users...